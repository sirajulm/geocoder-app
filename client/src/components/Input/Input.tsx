import React, { ChangeEvent } from 'react';
import styled from 'styled-components';


/* Returns a controlled Input component with label, if provided */

interface InputProps {
    name: string;
    label?: string;
    value: string;
    placeholder?: string
    required: boolean
    onChange: (event: ChangeEvent<HTMLInputElement>) => void
}

export const StyledLabel = styled.label`
    display: block;
    margin-bottom: 5px;
    font-family: 'Open Sans', sans-serif;
    font-weight: 600;
`;

export const FormGroup = styled.div`
    margin: 15px;
`;

export const StyledInput = styled.input`
    width: 100%;
    height: 40px;
    padding: 0 10px;
    font-family: 'Open Sans', sans-serif;
    font-weight: 400;
    font-size:16px;
    border: 1px solid #dadada;
    border-radius: 5px;
    outline: none;
`;

const Input = (props: InputProps) => {
    return (
        <FormGroup>
            {
                props.label && props.label.trim() !== ''
                && <StyledLabel
                    htmlFor={props.name} >
                    {props.label}{props.required ? ' *' : ''}
                </StyledLabel>
            }
            <StyledInput
                type="text"
                name={props.name}
                value={props.value}
                placeholder={props.placeholder}
                required={props.required || false}
                onChange={props.onChange}
            />
        </FormGroup>
    )
}

export default Input;