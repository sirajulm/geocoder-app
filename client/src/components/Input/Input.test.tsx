import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'
import 'jest-styled-components'


import Input, { StyledLabel, StyledInput } from './Input';

describe('Input', () => {
    describe('StyledLabel', () => {
        it('should render', () => {
            const component = renderer.create(<StyledLabel />).toJSON()
            expect(component).toMatchSnapshot()
        })
    })
    describe('StyledInput', () => {
        it('should render', () => {
            const component = renderer.create(<StyledInput />).toJSON()
            expect(component).toMatchSnapshot()
            expect(component).toHaveStyleRule('border', '1px solid #dadada')
        })
    })

    describe('InputComponent', () => {
        it('should render input', () => {
            const component = mount(<Input />)
            const label = component.find('label')
            const input = component.find('input')

            expect(label.length).toBe(0)
            expect(input.length).toBe(1)
        })
        it('should render label', () => {
            const label = 'Sample Label'
            const component = mount(<Input label={label} />)
            const labelElement = component.find('label')
            const inputElement = component.find('input')
            expect(labelElement.text()).toBe(label)
            expect(inputElement.props().type).toBe('text')
        })

        it('should render input required', () => {
            const label = 'Sample Label'
            const component = mount(<Input label={label} required={true} />)
            const labelElement = component.find('label')
            const inputElement = component.find('input')

            expect(labelElement.text()).toBe(`${label} *`)
            expect(inputElement.props().required).toBe(true)
        })

        it('should trigger onChange', () => {
            const onChangeMock = jest.fn();
            const value = 'Sample Value';

            const component = mount(<Input onChange={onChangeMock} value={value} />);
            const inputElement = component.find('input').simulate('change');
            expect(onChangeMock).toBeCalled()
            expect(onChangeMock.mock.calls[0][0].target.value).toBe(value)

        })

    })
})
