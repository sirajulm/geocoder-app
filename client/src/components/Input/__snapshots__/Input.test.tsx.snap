// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input StyledInput should render 1`] = `
.c0 {
  width: 100%;
  height: 40px;
  padding: 0 10px;
  font-family: 'Open Sans',sans-serif;
  font-weight: 400;
  font-size: 16px;
  border: 1px solid #dadada;
  border-radius: 5px;
  outline: none;
}

<input
  className="c0"
/>
`;

exports[`Input StyledLabel should render 1`] = `
.c0 {
  display: block;
  margin-bottom: 5px;
  font-family: 'Open Sans',sans-serif;
  font-weight: 600;
}

<label
  className="c0"
/>
`;
